src/utilities/updateBrowserUrl.ts

```diff
@@ -0,0 +1,39 @@
+import { isBrowser } from '@/utilities/isBrowser'
+
+type UpdateBrowserUrlOptions = {
+  replace?: boolean,
+}
+
+export function updateBrowserUrl(url: string, options: UpdateBrowserUrlOptions = {}): void {
+  if (!isBrowser) {
+    return
+  }
+
+  if (isSameOrigin(url)) {
+    return updateHistory(url, options)
+  }
+
+  return updateWindow(url, options)
+}
+
+function updateHistory(url: string, options: UpdateBrowserUrlOptions): void {
+  if (options.replace) {
+    return history.replaceState({}, '', url)
+  }
+
+  history.pushState({}, '', url)
+}
```

Comment:
again just calling out that this responsibility probably should/will get moved out in favor of something like `routerHistory.update(url, options)`

Reply:
Let's sync up on this. I didn't imagine history being in charge of updating the url. But just tracking history itself.